runtime/common/xwalk_resource_delegate.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2016 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_COMMON_XWALK_RESOURCE_DELEGATE_H_
+#define XWALK_RUNTIME_COMMON_XWALK_RESOURCE_DELEGATE_H_
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "base/macros.h"
+#include "base/memory/ref_counted_memory.h"
+#include "ui/base/resource/resource_bundle.h"
+
+namespace base {
+class FilePath;
+}
+
+namespace gfx {
+class Image;
+}
+
+namespace xwalk {
+
+// In Chrome,the locales path is set as "locales/" which potentially conflicts
+// with it is in Crosswalk. So purpose of XWalkResourceDelegate is to set the
+// locale pack path of Crosswalk as "locales/xwalk/".
+class XWalkResourceDelegate : public ui::ResourceBundle::Delegate {
+ public:
+  // Returns the singleton of delegate. It doesn't create an instance.
```

Comment:
Not only are you introducing a dead method that is never called (and thus the singleton pattern is never used), but the `DCHECK`s in the constructor and destructor means Crosswalk will crash every time in debug mode since `g_instance` will be null.